cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -535,28 +568,34 @@ class SnappyUnifiedMemoryManager private[memory](
   override def releaseStorageMemoryForObject(objectName: String,
                                              numBytes: Long,
                                              memoryMode: MemoryMode): Unit = synchronized {
-    logDebug(s"releasing [SNAP] memory for $objectName = $numBytes")
-    if (memoryForObject.addTo(objectName, -numBytes) >= 0L) {
+    logDebug(s"releasing $managerId memory for $objectName = $numBytes")
+    if (memoryForObject.addTo(objectName -> memoryMode, -numBytes) != 0L) {
```

Comment:
Isn't >= 0 somewhat safer just in case there is an accounting mistake and results goes into -ve!